.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,56 @@
+name: Publish to npm
+
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  cache-build:
+    permissions:
+      contents: write
+    if: |
+      github.event.pull_request.merged == true &&
+      startsWith(github.event.pull_request.head.ref, 'release/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # We check out the release pull request's base branch, which will be
+          # used as the base branch for all git operations.
+          ref: ${{ github.event.pull_request.base.ref }}
+      - name: Get Node.js version
+        id: nvm
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
+      - run: |
+          yarn setup
+          yarn build
+      - uses: actions/cache@v2
+        id: restore-build
+        with:
+          path: ./*
+          key: ${{ github.sha }}
+
+  publish-npm:
+    name: Publish to npm
+    runs-on: ubuntu-latest
+    needs: cache-build
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get Node.js version
+      id: nvm
+      run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
+    - uses: actions/cache@v2
+      id: restore-build
+      with:
+        path: ./*
+        key: ${{ github.sha }}
+    - name: Publish
+      uses: rickycodes/action-npm-publish@c561c6562c3f8c24cd6372b526a5feb138506332
+      with:
+        npm-token: ${{ secrets.NPM_TOKEN }}
```

Comment:
this will need to be setup with an NPM automation token. ideally we make this org wide so we can easily consume the same secret in other repos where we want to publsih automatically from.